### PR TITLE
Config: Expose "DisableScreenSaver" in the UI and default it to true.

### PR DESCRIPTION
**Pull request by jordan-woyak**
"Inhibit Screensaver During Emulation"
image

### CODE REVIEW

**Comment by Tilka** [comment]
The "while running" makes it sound like the screensaver is only inhibited while a game is running.

**Comment by MayImilae** [comment]
I assume this also inhibits power saving modes such as turning off the display and sleep?

Also, does this only apply while a game is running, or always if Dolphin is running?

**Comment by Tilka** [comment]
>I assume this also inhibits power saving modes such as turning off the display and sleep?

Depends on the platform.

>Also, does this only apply while a game is running, or always if Dolphin is running?

Only while a game is running. Need to clarify the wording imo.

**Comment by mbc07** [comment]
An issue that I often have on Windows 10 is that while Dolphin prevents the screen from turning off and the computer from entering sleep mode when this option is enabled, if you're playing exclusively with controllers, the Windows 10 Lock Screen will still come up after sometime, and when that happens it generally also makes Dolphin exit fullscreen mode.

Does this PR do anything to improve on that (e.g. preventing also the lock screen) or it just exposes the existing option with the existing behavior in the GUI?

**Comment by jordan-woyak** [comment]
@mbc07 This just exposes the existing option.

**Comment by 8times9** [comment]
What about a more commonly-used verb like "Prevent Screen Saver While Running"?

**Comment by leoetlino** [comment]
What's the status of this PR?

**Comment by jordan-woyak** [comment]
Does everyone like May's suggested: "Inhibit Screensaver While a Game is Running"?

**Comment by leoetlino** [comment]
Personally I'm fine with it -- the only change to the wording that needed to be done is to clarify that the option only applies while a game is running.

**Comment by leoetlino** [comment]
Ping ^^

**Comment by gingerbeardman** [comment]
Just had to Google for what to edit in the .ini so this option is much needed. Thanks

File: Source/Core/DolphinQt/Settings/InterfacePane.cpp [hunk 1]
```diff
@@ -151,11 +152,13 @@ void InterfacePane::CreateUI()
   m_checkbox_use_covers =
       new QCheckBox(tr("Download Game Covers from GameTDB.com for Use in Grid Mode"));
   m_checkbox_show_debugging_ui = new QCheckBox(tr("Show Debugging UI"));
+  m_checkbox_disable_screensaver = new QCheckBox(tr("Inhibit Screensaver While Running"));
```
[the remark this answers is not in the document]
Comment (Rumi-Larry) [reply Source/Core/DolphinQt/Settings/InterfacePane.cpp]: Perhaps "Inhibit sleep mode during emulation"

[the remark this answers is not in the document]
Comment (jordan-woyak) [reply Source/Core/DolphinQt/Settings/InterfacePane.cpp]: I do like "During Emulation" better than "While a Game is Running".

**Comment by JosJuice** [comment]
If you change the setting while a game is running, does the change apply without having to restart?

**Comment by gingerbeardman** [comment]
@JosJuice did you try it? If so what did you see?

**Comment by jordan-woyak** [comment]
@JosJuice You'd have to at least pause/unpause to see the setting take effect. I suppose ideally that would not be the case. :P

**Review by leoetlino — APPROVED** [review]

**Pull request merged**